include/boost/uuid/detail/random_provider_getrandom.ipp

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2018 Andrey Semashev
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+//   http://www.boost.org/LICENCE_1_0.txt)
+//
+// getrandom() capable platforms
+//
+
+#include <boost/config.hpp>
+#include <boost/throw_exception.hpp>
+#include <boost/predef/library/c/gnu.h>
+#include <cerrno>
+#include <cstddef>
+
+#if !defined(BOOST_UUID_RANDOM_PROVIDER_GETRANDOM_IMPL_GETRANDOM)
+#if BOOST_LIB_C_GNU >= BOOST_VERSION_NUMBER(2, 25, 0)
+#define BOOST_UUID_RANDOM_PROVIDER_GETRANDOM_HAS_LIBC_WRAPPER
+#elif defined(__has_include)
+#if __has_include(<sys/random.h>)
```

Comment:
I've never seen __has_include before.  I was just thinking about how nice it would be to have the following include syntax in C++:

```
#include <boost/uuid/sha1.hpp> or <boost/uuid/detail/sha1.hpp>
```

To handle headers that move without leaving behind forwarding headers.

Reply:
>I've never seen __has_include before.

It's one of the SD-6 macros most recent compilers support.

https://en.cppreference.com/w/cpp/experimental/feature_test